go/test/endtoend/vtgate/tablet_healthcheck/reparent_test.go

```diff
@@ -0,0 +1,243 @@
+	"flag"
+	"fmt"
+	"os"
+	"sync"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+
+	"vitess.io/vitess/go/mysql"
+	"vitess.io/vitess/go/test/endtoend/cluster"
+)
+
+var (
+	clusterInstance *cluster.LocalProcessCluster
+	vtParams        mysql.ConnParams
+
+	// tabletRefreshInterval is the interval at which the tablet health check will refresh.
+	// This value is set to a high value to ensure that the vtgate does not attempt to refresh the tablet between the time a tablet is added and the time it is promoted.
+	tabletRefreshInterval = time.Hour
+
+	keyspaceName = "healthcheck_test_ks"
+	cell         = "healthcheck_test_cell"
+	shards       = []string{"-80", "80-"}
+	schemaSQL    = `
+create table customer(
+	customer_id bigint not null auto_increment,
+	email varbinary(128),
+	primary key(customer_id)
+) ENGINE=InnoDB;
+create table corder(
+	order_id bigint not null auto_increment,
+	customer_id bigint,
+	sku varbinary(128),
+	price bigint,
+	primary key(order_id)
+) ENGINE=InnoDB;
+`
+
+	vSchema = `
+{
+	"sharded": true,
+	"vindexes": {
+		"hash": {
+			"type": "hash"
+		}
+	},
+	"tables": {
+		"customer": {
+			"column_vindexes": [
+				{
+					"column": "customer_id",
+					"name": "hash"
+				}
+			]
+		},
+		"corder": {
+			"column_vindexes": [
+				{
+					"column": "customer_id",
+					"name": "hash"
+				}
+			]
+		}
+	}
+}
+`
+)
+
+// TestMain sets up the vitess cluster for any subsequent tests
+func TestMain(m *testing.M) {
+	defer cluster.PanicHandler(nil)
+	flag.Parse()
+
+	exitCode := func() int {
+		clusterInstance = cluster.NewCluster(cell, "localhost")
+		defer clusterInstance.Teardown()
+
+		// Start topo server
+		err := clusterInstance.StartTopo()
+		if err != nil {
+			return 1
+		}
+
+		// Start keyspace
+		keyspace := &cluster.Keyspace{
+			Name:      keyspaceName,
+			SchemaSQL: schemaSQL,
+			VSchema:   vSchema,
+		}
+		clusterInstance.VtTabletExtraArgs = append(clusterInstance.VtTabletExtraArgs, []string{"--health_check_interval", "1s"}...)
+		err = clusterInstance.StartKeyspace(*keyspace, shards, 0, false)
+		if err != nil {
+			return 1
+		}
+
+		clusterInstance.VtGateExtraArgs = append(clusterInstance.VtGateExtraArgs, []string{"--tablet_refresh_interval", tabletRefreshInterval.String()}...)
+		err = clusterInstance.StartVtgate()
+		if err != nil {
+			return 1
+		}
+
+		vtParams = mysql.ConnParams{
+			Host: clusterInstance.Hostname,
+			Port: clusterInstance.VtgateMySQLPort,
+		}
+		return m.Run()
+	}()
+	os.Exit(exitCode)
+}
+
+// TestHealthCheckExternallyReparentNewTablet ensures that calling TabletExternallyReparented on a new tablet will switch the primary tablet
+// without having to wait for the tabletRefreshInterval.
+func TestHealthCheckExternallyReparentNewTablet(t *testing.T) {
+	ctx := context.Background()
+
+	// verify output of `show vitess_tablets` and `INSERT` statement
+	vtgateConn, err := mysql.Connect(ctx, &vtParams)
+	require.Nil(t, err)
```

Comment:
Nit, but `require.NoError(t, err)` is more widely used and more explicit.

Reply:
Fixed by [f2c1c7d](https://github.com/vitessio/vitess/pull/12893/commits/f2c1c7d716ad1de724ad4d676c26213a8cf1a3ed).